allennlp/training/metrics/evalb_bracketing_scorer.py

```diff
@@ -0,0 +1,127 @@
+    on parse trees. Here, we use it to compute the Precision, Recall and F1 metrics.
+    You can download the source for EVALB from here: <http://nlp.cs.nyu.edu/evalb/>.
+
+    Note that this software is 20 years' old. In order to compile it on modern hardware,
+    you may need to remove an ``include <malloc.h>`` statement in ``evalb.c`` before it
+    will compile.
+
+    AllenNLP contains the EVALB software, but you will need to compile it yourself
+    before using it because the binary it generates is system depenedent. To build it,
+    run ``make`` inside the ``scripts/EVALB`` directory.
+
+    Parameters
+    ----------
+    evalb_directory_path : ``str``, required.
+        The directory containing the EVALB executable.
+    evalb_param_filename: ``str``, optional (default = "COLLINS.prm")
+        The relative name of the EVALB configuration file used when scoring the trees.
+        By default, this uses the COLLINS.prm configuration file which comes with EVALB.
+    """
+    def __init__(self, evalb_directory_path: str, evalb_param_filename: str = "COLLINS.prm") -> None:
+        self._evalb_program_path = os.path.join(evalb_directory_path, "evalb")
+        self._evalb_param_path = os.path.join(evalb_directory_path, evalb_param_filename)
+
+        if not os.path.exists(self._evalb_param_path):
+            raise ConfigurationError("You must compile the EVALB scorer before using it."
+                                     " Run 'make' in the 'scripts/EVALB' directory.")
+
+        self._recall_regex = re.compile(r"Bracketing Recall\s+=\s+(\d+\.\d+)")
+        self._precision_regex = re.compile(r"Bracketing Precision\s+=\s+(\d+\.\d+)")
+        self._f1_measure_regex = re.compile(r"Bracketing FMeasure\s+=\s+(\d+\.\d+)")
+
+        self._precision = 0.0
+        self._recall = 0.0
+        self._f1_measure = 0.0
+        self._count = 0.0
+
+    @overrides
+    def __call__(self, predicted_trees: List[Tree], gold_trees: List[Tree]) -> None: # type: ignore
+        """
+        Parameters
+        ----------
+        predicted_trees : ``List[Tree]``
+            A list of predicted NLTK Trees to compute score for.
+        gold_trees : ``List[Tree]``
+            A list of gold NLTK Trees to use as a reference.
+        """
+        tempdir = tempfile.gettempdir()
+        gold_path = os.path.join(tempdir, "gold.txt")
+        predicted_path = os.path.join(tempdir, "predicted.txt")
+        output_path = os.path.join(tempdir, "output.txt")
+        with open(gold_path, "w") as gold_file:
+            for tree in gold_trees:
+                gold_file.write(f"{tree}\n")
+
+        with open(predicted_path, "w") as predicted_file:
+            for tree in predicted_trees:
+                predicted_file.write(f"{tree}\n")
+
+        command = f"{self._evalb_program_path} -p {self._evalb_param_path} " \
+                  f"{gold_path} {predicted_path} > {output_path}"
+        subprocess.run(command, shell=True, check=True)
```

Comment:
You could consider just grabbing `stdout` directly from the command (search for "python subprocess pipe"), instead of redirecting the output to a file and reading the file.  One less trip to and from disk.